### ISSUE TITLE
Dataset availability

### ISSUE THREAD

**Issue by gus-morales** [8values/8values.github.io#29]
It would be nice to have a handle on the dataset, and see how each of the results compare along where you live, study, education level, etc. Me and a friend would gladly submit a code with statistical analysis to work on something along these lines. Is it possible to add an option to download the latest dataset?

**Comment by ommirandap** [comment]
I strongly agree that there could be an interesting opportunity to collect important data here.
And it shouldn't be so difficult, after the poll, you can ask politely to the user if want to share their:
Age, Country, Gender (maybe), educational status, etc...
(actually, maybe you can find something interesting with google analytics data)

**Comment by TristanBomb** [comment]
While I agree this would be a great thing to have, especially to help balance the test, I'm not currently collecting data.

**Comment by urbanpat** [comment]
One solution would be to pull the server logs to see what final results people end up with, since they're included in the URL. Should be possible to post a text file of all the result URLs called and let people extract individual parameters for analysis.

Of course you'd have to accept some level of error from people like me who adjusted results until I got Nazism to scare my friends...

**Comment by mitar** [comment]
Ehm, what about those who do not want to have their data stored? Maybe a better way would be to ask at the end if they want to contribute their data to a common public dataset and in that case data gets submitted to the server. Opt-in.

**Comment by gus-morales** [comment]
In that case I vote for an opt-out approach.

**Comment by mitar** [comment]
I really do think it should be opt-in. This is to empower people to make a decision. We have already enough tracing on the web and guessing of my preferences.

**Comment by gus-morales** [comment]
I agree with your attitude towards tracing, but I assume the dev is not looking to sell that data, or to participate in marketing. These would be used for simple statistics, which in turn would go back to the general public.

**Comment by mitar** [comment]
Still. Even if you do an academic research on human subjects, that has to be opt-in with full consent f a person. Otherwise you cannot publish a paper. If you really want to make this dataset full useful, we should follow similar guidelines.

**Comment by urbanpat** [comment]
I think you misunderstand. When you visit websites your URL requests and IP address are logged by the webserver. You can use VPNs to anonymize your IP, but you cannot avoid having your requests logged because the server must know what pages to send you.

**Comment by gus-morales** [comment]
Also, there are *tons* of useful datasets that does not follow similar guidelines. Similarly, the opposite is also true. Tracing has more to do with ethics, not with how useful the dataset is. And website cookies do this anyway already.

**Comment by mitar** [comment]
Not necessary, if it is single page app where client-side changes the URL. This seems to be the case of this site at the moment. Furthermore, it is hosted on GitHub, so any URL requests logs are on GitHub.

**Comment by urbanpat** [comment]
Yes, that's true.

**Comment by mitar** [comment]
So we would on purpose want to change this so that users would be more traceable? Why would that be a good thing. On the other hand, we could make it opt-in, when JavaScript submits the results to a server for those who do want to participate in the research. I have done similar things in the past and many users at the end do like to contribute and help, especially if you tell them that they can also get the data. I think this is much better proposition.

**Comment by gus-morales** [comment]
Because to me (the guy who originally proposed to work on this dataset), this is not about tracing people. My code would not trace anyone nor I would sell the data to third parties.

Now, your second argument seems more appropriate to the discussion. I was suggesting an opt-out approach which I assume would yield more data (e.g. Dan Ariely's TED talk on decision making), but of course I am open to personal experiences on that matter.

**Comment by mitar** [comment]
Yes, opt-out would probably get more data, but I am saying that in this case opt-in will probably also give you enough data, but in a more ethical way. Just because you can do something it does not mean you should be doing something.

**Comment by urbanpat** [comment]
There is nothing unethical about collecting anonymous data provided to a
website in a form request.

**Comment by mitar** [comment]
If you inform users at the beginning that you will be doing so, true.

**Comment by gus-morales** [comment]
Exactly. Also, this has becoming lecturing and I prefer to stay out of it. All I am saying is that collecting data is not necessarily unethical, specially if you tell them before hand. Yes, that *could* lead to tracing and that I agree, *might* be unethical, but hardly the point of this post.

**Comment by urbanpat** [comment]
Collecting data is not, on its own, unethical. In fact, as I pointed out
above, it's completely inevitable. Every server you ping keeps a log. It's
very silly to claim it's unethical to track which buttons your anonymous
users click on. That's like saying it's unethical to use web counters on
your geocities page in 1996.

**Comment by mitar** [comment]
That is why websites have privacy policies where they explain what they are collecting. Even if they are collecting just HTTP logs. And no, you do not have to be collecting HTTP logs. You might decide to collect them, maybe you did it automatically, because everyone else is collecting them. But it is still a decision you made. And no, not all websites are collecting them, especially for the privacy reasons.

And sites which collect physiological data like this website should keep such decisions in mind. It really depends on what you are collecting. Here, data being collected is highly personal data.

**Comment by mitar** [comment]
A web counter does not reveal your personality. This site here tries to do exactly that. You do not see the difference?

**Comment by gus-morales** [comment]
Well yes, a privacy policy I agree is totally on point here. That and a notification of data collection. But that alone does not make the act of collecting *anonymous* data alone "less ethical"; it's just an act of transparency.

**Comment by gus-morales** [comment]
Also, as long as it is anonymous data I don't see it as "highly personal", since you cannot relate this information to any person (like tracing does). The fact that the algorithm is trying to predict personality traits does not make it "more personal".

**Comment by mitar** [comment]
Yes, this is one approach. The other is to give them an option (opt-in) to do a consensual decision to be part of this. You said yourself that you know that many people will probably just leave the default option. And why is that, why is not opt-in and opt-out the same? Because of those people who do not read the privacy policy, who do not think about potential implications. That is why it is important to make good defaults.

And no. Opt-in and opt-out do not have same ethical value. Opt-in is more ethical. Because you are not assuming that people will not have an issue collecting highly personal data.

**Comment by mitar** [comment]
> Also, as long as it is anonymous data I don't see it as "highly personal", since you cannot relate this information to any person (like tracing does).

By itself not. But the site also calls to statcounter.com. So then combining statcounter.com calls with some other calls to statcounter.com where you maybe reveal your identity also deanonymize those users.

**Comment by gus-morales** [comment]
As long as they can read (which I am assuming they can), an opt-out approach with a notification is consensual.

And if you want to keep discussing this please open another thread. Both approaches are fairly clear by now, and I greatly appreciate your feedback.

**Comment by mitar** [comment]
I think the whole academic world who is dealing with human subjects would disagree with you here. Just Google.

For example: http://www.ethicsguidebook.ac.uk/Opt-in-and-opt-out-sampling-94

**Comment by gus-morales** [comment]
" ‘Opt-out’ samples are those where participants are contacted *without volunteering* to take part in the research..."
If you don't see the difference between that definition and what I have suggested, I have not much to say.

**Comment by mitar** [comment]
I think you should read a bit more. What I think you are saying (can you point me to exactly where are you describing the protocol the participants would follow) is a pretty standard definition of opt-out. Everyone doing the quiz will have their data collected unless they say that they do not want that. That is opt-out.

**Comment by gus-morales** [comment]
Yes but the "without volunteering" part is key to understand my point of view, assuming there is a public policy and a notification of collection.

And honestly we are splitting hairs now. The idea I originally suggested would have zero impact on the current status of privacy on the Internet, in either direction.

**Comment by mitar** [comment]
Yes, by using the website I do not volunteer to participate in collecting of data. I have not **freely give** my consent.

**Comment by elifoster** [comment]
I agree it should be opt in.

**Comment by TristanBomb** [comment]
Alright, so this discussion seems to have run its course, but having concrete data would absolutely be useful for reworking ideology.

As for opt-in vs opt-out: I'm in favor of opt-out, personally. As long as it's clear that the results will be collected, _but_ you can opt out, I don't see an ethical problem with it. You're still getting permission, the data is anonymous, and the data would not be sold to anyone. (If I wanted to make money with this project, I would put advertisements on it.)

**Comment by Berkmann18** [comment]
@TristanBomb To collect data (assuming the user agreed to it), you could store those in a DB (say, MongoDB) and make it available for those who would like to analyse the results (say for research purposes).

**Comment by BradKML** [comment]
Opt-in: "participation bias" ruining data, which can be handled through 
Opt-out: ethics regarding "explicit consent" regarding privacy

Is that correct?
